eng/scripts/automation/Invoke-GenerateAndBuild.ps1

```diff
@@ -41,27 +42,59 @@ if ($autorestConfig -ne "") {
     }
 }
 
-Write-Host "swaggerDir:$swaggerDir, readmeFile:$readmeFile"
+$generatedSDKPackages = New-Object 'Collections.Generic.List[System.Object]'
 
 # $service, $serviceType = Get-ResourceProviderFromReadme $readmeFile
 $sdkPath =  (Join-Path $PSScriptRoot .. .. ..)
 $sdkPath = Resolve-Path $sdkPath
 $sdkPath = $sdkPath -replace "\\", "/"
 
-$readme = ""
-if ($commitid -ne "") {
-  if ($repoHttpsUrl -ne "") {
-    $readme = "$repoHttpsUrl/blob/$commitid/$readmeFile"
+if ($readmeFile) {
+  Write-Host "swaggerDir:$swaggerDir, readmeFile:$readmeFile"
+
+  $readme = ""
+  if ($commitid -ne "") {
+    if ($repoHttpsUrl -ne "") {
+      $readme = "$repoHttpsUrl/blob/$commitid/$readmeFile"
+    } else {
+      $readme = "https://github.com/$org/azure-rest-api-specs/blob/$commitid/$readmeFile"
+    }
   } else {
-    $readme = "https://github.com/$org/azure-rest-api-specs/blob/$commitid/$readmeFile"
+    $readme = (Join-Path $swaggerDir $readmeFile)
   }
-} else {
-  $readme = (Join-Path $swaggerDir $readmeFile)
+  Invoke-GenerateAndBuildSDK -readmeAbsolutePath $readme -sdkRootPath $sdkPath -autorestConfigYaml "$autorestConfigYaml" -downloadUrlPrefix "$downloadUrlPrefix" -generatedSDKPackages $generatedSDKPackages
```

Comment:
Nit: you don't need to quote variables when they are the only parameter value.

Reply:
Thanks @heaths . I removed the unused quote.